### PR TITLE
*: instead scheduler type string of types.CheckerSchedulerType

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/8379

### What is changed and how does it work?

```commit-message
- Remove the old scheduler type, such as BalanceLeaderType
- Instead old scheduler type string of types.CheckerSchedulerType
- Instead old scheduler name of types.CheckerSchedulerType in SchedulerRegisterMap
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/coordinator.go [hunk 1]
```diff
@@ -275,7 +276,7 @@ func (c *Coordinator) InitSchedulers(needRun bool) {
 		typ := schedulers.FindSchedulerTypeByName(name)
 		var cfg sc.SchedulerConfig
 		for _, c := range scheduleCfg.Schedulers {
-			if c.Type == typ {
+			if c.Type == types.SchedulerTypeCompatibleMap[typ] {
```
Comment (okJiang) [inline pkg/schedule/coordinator.go:279]: How about renaming to `ConvertNewType2OldStr`

File: pkg/schedule/type/type.go [hunk 2]
```diff
@@ -73,48 +73,75 @@ const (
...
+		"shuffle-hot-region":      ShuffleHotRegionScheduler,
+		"shuffle-leader":          ShuffleLeaderScheduler,
+		"shuffle-region":          ShuffleRegionScheduler,
+		"split-bucket":            SplitBucketScheduler,
+		"transfer-witness-leader": TransferWitnessLeaderScheduler,
+		"label":                   LabelScheduler,
+	}
+
+	// SchedulerStr2Type is a map to convert the scheduler string to the CheckerSchedulerType.
+	SchedulerStr2Type = map[string]CheckerSchedulerType{
```
Comment (rleungx) [inline pkg/schedule/type/type.go:127]: StringToSchedulerType

**Review by rleungx — APPROVED** [review]
The rest LGTM

File: pkg/mcs/scheduling/server/cluster.go [hunk 3]
```diff
@@ -309,11 +309,12 @@ func (c *Cluster) updateScheduler() {
 		)
 		// Create the newly added schedulers.
 		for _, scheduler := range latestSchedulersConfig {
+			schedulerType := types.ConvertOldStr2Type[scheduler.Type]
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/cluster.go:312]: How does ConvertOldStrToType sound to you? Variable names with embedded numbers are not common in PD.

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: pkg/schedule/schedulers/balance_leader.go [hunk 4]
```diff
@@ -43,8 +43,6 @@ import (
 const (
 	// BalanceLeaderName is balance leader scheduler name.
 	BalanceLeaderName = "balance-leader-scheduler"
```
[the remark this answers is not in the document]
Comment (okJiang) [reply pkg/schedule/schedulers/balance_leader.go:45]: It will be in the next pr

File: pkg/mcs/scheduling/server/cluster.go (hunk 3 above)
Comment (okJiang) [reply pkg/mcs/scheduling/server/cluster.go:312]: They are same for me. I can update it

**Review by niubell — APPROVED** [review]

**Pull request merged**